app/models/spree/order.rb

```diff
@@ -0,0 +1,915 @@
+# frozen_string_literal: true
+
+require 'spree/order/checkout'
+require 'open_food_network/enterprise_fee_calculator'
+require 'open_food_network/feature_toggle'
+require 'open_food_network/tag_rule_applicator'
+require 'concerns/order_shipment'
+
+ActiveSupport::Notifications
+  .subscribe('spree.order.contents_changed') do |_name, _start, _finish, _id, payload|
+  payload[:order].reload.update_distribution_charge!
+end
+
+module Spree
+  class Order < ActiveRecord::Base
+    prepend OrderShipment
+    include Checkout
+
+    checkout_flow do
+      go_to_state :address
+      go_to_state :delivery
+      go_to_state :payment, if: ->(order) {
+        order.update_totals
+        order.payment_required?
+      }
+      go_to_state :complete
+      remove_transition from: :delivery, to: :confirm
```

Comment:
now that we removed the :confirm state this line can be removed.

Reply:
good catch :+1: done